packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ICXGenericResult, ICXOfferInfo, ICXOrderInfo, ICXOffer, ICXOrder, ICXOrderStatus, ICXOrderType, UTXO } from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import { ICXSetup, symbolDFI, accountDFI, idDFI, accountBTC, ICX_TAKERFEE_PER_BTC, DEX_DFI_PER_BTC_RATE } from './icx_setup'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('ICXOrderBook.makeOffer', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  const icxSetup = new ICXSetup(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await icxSetup.createAccounts()
+    await icxSetup.createBTCToken()
+    await icxSetup.initializeTokensIds()
+    await icxSetup.mintBTCtoken(100)
+    await icxSetup.fundAccount(accountDFI, symbolDFI, 500)
+    await icxSetup.fundAccount(accountBTC, symbolDFI, 10) // for fee
+    await icxSetup.createBTCDFIPool()
+    await icxSetup.addLiquidityToBTCDFIPool(1, 100)
```

Comment:
From an issue in ain, when `addLiquidityToBTCDFIPool` is called between order and swap, if price moved (add/remove/swap), it causes price fluctuation and the order to fail.

Reply:
yes. plan to add a test for that in complex tests. likewise, there could be some scenarios remaining, which I would try to add to the complex tests as we go.